src/Pelagos/Util/DataStore.php

```diff
@@ -424,4 +410,26 @@ protected function getStoreFileName($datasetId, $type)
         }
         return $storeFileName;
     }
+
+    /**
+     * Get the numeric ID of a POSIX user or group.
+     *
+     * @param string  $name    The user or group name.
+     * @param boolean $isGroup Flag set to true if a group, defaults to user.
+     *
+     * @return string|boolean
+     */
+    protected function getIdFromName(string $name, bool $isGroup = false)
+    {
+        $id = false;
+        if ($isGroup) {
+            // This call returns false on failure to resolve.
+            $obj = posix_getgrnam($name);
+            $id = $obj['gid'];
+        } else {
+            $obj = posix_getpwnam($name);
+            $id = $obj['uid'];
+        }
+        return $id;
```

Comment:
Can id also have a string value?

Reply:
no, just either a numeric value or a boolean false as posix_getpwnam and the other return false on failure. Come to think of it, it's probably better I throw an exception if that were to ever happen.

Reply:
Oh then, we should probably check if array exists before trying to access the array $obj